qcodes/instrument_drivers/Minicircuits/USB_SPDT.py

```diff
@@ -0,0 +1,88 @@
+from qcodes import Instrument
+from qcodes.utils import validators as vals
+from qcodes.instrument.channel import InstrumentChannel, ChannelList
+
+
+class SwitchChannelUSB(InstrumentChannel):
+    def __init__(self, parent, name, channel_letter):
+        """
+        Args:
+            parent (Instrument): The instrument the channel is a part of
+            name (str): the name of the channel
+            channel_letter (str): channel letter ['a', 'b', 'c' or 'd'])
+        """
+
+        super().__init__(parent, name)
+        self.channel_letter = channel_letter.upper()
+        _chanlist = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
+        self.channel_number = _chanlist.index(channel_letter)
+
+        self.add_parameter('switch',
+                           label='switch',
+                           set_cmd=self._set_switch,
+                           get_cmd=self._get_switch,
+                           vals=vals.Ints(1, 2)
+                           )
+
+    def _set_switch(self, switch): 
+        self._parent.switch.Set_Switch(self.channel_letter,switch-1)
+
+    def _get_switch(self):
+        status = self._parent.switch.GetSwitchesStatus(self._parent.address)[1]
+        return int("{0:04b}".format(status[-1-self.channel_number]))+1
```

Comment:
Status should be cast to binary form before picking the number connected to the channel (moving one parenthesis):
int("{0:04b}".format(status)[-1-self.channel_number])+1

Reply:
that must have been a typo...